### PR TITLE
update code to draft-tschofenig-rats-psa-token-08

### PR DESCRIPTION
**Pull request by thomas-fossati**

### CODE REVIEW

**Comment by thomas-fossati** [comment]
This PR partly addresses #9 .  The missing bit being implementing the [backwards compatibility scheme](https://www.ietf.org/archive/id/draft-tschofenig-rats-psa-token-08.html#name-backwards-compatibility-con).

**Comment by yogeshbdeshpande** [comment]
> This PR partly addresses #9 . The missing bit being implementing the [backwards compatibility scheme](https://www.ietf.org/archive/id/draft-tschofenig-rats-psa-token-08.html#name-backwards-compatibility-con).

Glad I read this now. I was reviewing and was wondering how the old claims is supported! :)

Now realized that bit is a WIP...

File: psatoken.go [hunk 1]
```diff
@@ -384,7 +405,9 @@ func (p *Claims) validateNonce() error {
 		return fmt.Errorf("missing mandatory nonce")
 	}
 
-	if err := isPSAHashType(*p.Nonce); err != nil {
+	nonce := ([]eat.Nonce)(*p.Nonce)[0]
```
Comment (yogeshbdeshpande) [inline psatoken.go:408]: Could the nonce decoding use a consistent type as used as Nonces on line 54 ?

File: psatoken_test.go [hunk 2]
```diff
@@ -426,13 +393,21 @@ func TestClaims_sign_and_verify(t *testing.T) {
 }
 
 func makeClaims() Claims {
-	profile := PSA_PROFILE_1
+	profile, _ := eat.NewProfile(PSA_PROFILE_2)
```
Comment (yogeshbdeshpande) [inline psatoken_test.go:396]: Shall we not test for error returned from eat.NewProfile()?

this way any incompatibility introduced can be caught...

File: psatoken.go (hunk 1 above)
Comment (thomas-fossati) [reply psatoken.go:408]: I need the pending changes in the eat repo (specifically, the new `GetI()` interface) to deal with this in a clean way.

File: psatoken_test.go (hunk 2 above)
Comment (thomas-fossati) [reply psatoken_test.go:396]: sgtm, will do!

Comment (thomas-fossati) [reply psatoken_test.go]: Done in b4b3f31

**Review by yogeshbdeshpande — APPROVED** [review]

**Pull request merged**